internal/proposals/improvement-of-log-messages-usability.md

```diff
@@ -0,0 +1,98 @@
+# Improvement of log messages usability
+
+> TODO: Ask TWs for a language review
+
+## Motivation
+
+We want to provide a common, unified solution to logging across Kyma components, which will enable users and maintainers to easily track specific events happening in Kyma's ecosystem. The solution should be independent of the platform (AWS, Azure, GCP) and easy to introduce in existing components
+
+Users, developers, and maintainers want to have an easy way to track events happening in Kyma, problems that occur, and debug faulty components. This will make understanding of Kyma's ecosystem easier and more transparent
+
+## Solution regarding the log messages
+
+Every mission-critical component (or a critical part of the component) must produce log entries in order to make the problem diagnosis process quick and effective
+
+Key assumptions for the log messages:
+
+- Transparency - every entry must be as short as possible but containing all essential information about the process
+- Tracing - entries sequence must be connected so there is an easy way to filter logs concerning a specific operation
+- GDPR-compliance - entries must not contain any sensitive data (password, personal data in the meaning of GDPR)
+- Agile - log messages and their context should be refactored. If the log was useful while debugging the incident, make other also that useful. If there was something missing in the log, report it and refactor
+- No "responsibility overlapping" - Because there are other tools to help in diagnosis, logs, in general, should not contain the information which can be taken from that tools (i.e there's tracing which can be used for timings)
+    > TODO: Ask Artur and Przemek for the review on the potential responsibility overlaping
+
+Additionally, we need an approach to the "external" components used in Kyma (the ones not implemented in the Kyma team). We should have an idea on how to approach this problem and present the potential solution
+
+## Log format
+
+To unify the logs, therefore to make debugging process much easier and to make logs parsing also super easy, I'd like to propose a single log format so every service, job and all the components could follow:
+
+- timestamp - ISO 8601 Date and time with timezone. For example: "2020-12-08T10:33:45+00:00"
```

Comment:
Worth notice that in Go, the RFC3339 is used, which is a stricter version of ISO 8601.
Generally those standard are very close to each other.
Here is a  detailed discussion on stackoverflow, which can help you
https://stackoverflow.com/questions/522251/whats-the-difference-between-iso-8601-and-rfc-3339-date-formats

Reply:
That's nice! I added it to the proposal